src/main/proto/th2_grpc_common/common.proto

```diff
@@ -200,8 +200,20 @@ message Checkpoint {
   string id = 1;
   map<string, DirectionCheckpoint> session_alias_to_direction_checkpoint = 2;
 
+  message CheckpointData {
+    int64 sequence = 1; // Message sequence number in session
+    google.protobuf.Timestamp timestamp = 2; // Message creation timestamp
+  }
+
   message DirectionCheckpoint {
-    map<int32, int64> direction_to_sequence = 1;
+    /*
+    * It should contain one of these two fields - direction_to_sequence or direction_to_checkpoint_data, preferably the second one.
+    * If you set both of them, in check1 will be generated an error.
+    * If direction_to_sequence is filled, it will be converted to direction_to_checkpoint_data.
+    * This field will be removed in future major releases. Please use direction_to_checkpoint_data instead.
+    */
+    map<int32, int64> direction_to_sequence = 1 [deprecated = true];
+    map<int32, CheckpointData> direction_to_checkpoint_data = 2;
```

Comment:
I believe we can use `oneof` here to prevent setting both `direction_to_sequence` and `direction_to_checkpoint_data`